packages/commonwealth/client/scripts/views/modals/ArchiveThreadModal.tsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import 'modals/ArchiveThreadModal.scss';
+import type Thread from '../../models/Thread';
+import app from 'state';
+import { CWButton } from '../components/component_kit/cw_button';
+import { CWText } from '../components/component_kit/cw_text';
+import { useEditThreadMutation } from 'state/api/threads';
+
+import {
+  notifyError,
+  notifySuccess,
+} from '../../controllers/app/notifications';
+
+type ArchiveThreadModalProps = {
+  onModalClose: () => void;
+  thread: Thread;
```

Comment:
Just to confirm, Is this a presentation component? If not, I feel like `thread: Thread;` is too broad, can we pass specific fields?

Reply:
It's a presentational component.